nanoFramework.System.Net.Http/Http/System.Net.HttpWebRequest.cs

```diff
@@ -73,38 +73,47 @@ public class HttpWebRequest : WebRequest
         /// <param name="unused">Unused</param>
         static private void CheckPersistentConnections(object unused)
         {
-            int count = m_ConnectedStreams.Count;
-            // The fastest way to exit out - if there are no sockets in the list - exit out.
-            if (count > 0)
+            // skigrinder - 2020-12-10 added exception handling & fixed 'timePassed' setting
+            try
             {
-                DateTime curTime = DateTime.UtcNow;
-                lock (m_ConnectedStreams)
+                int count = m_ConnectedStreams.Count;
+                // The fastest way to exit out - if there are no sockets in the list - exit out.
+                if (count > 0)
                 {
-                    count = m_ConnectedStreams.Count;
-
-                    for (int i = count-1; i >= 0; i--)
+                    DateTime curTime = DateTime.UtcNow;
+                    lock (m_ConnectedStreams)
                     {
-                        InputNetworkStreamWrapper streamWrapper = (InputNetworkStreamWrapper)m_ConnectedStreams[i];
-
-                        TimeSpan timePassed = streamWrapper.m_lastUsed - curTime;
-
-                        // If the socket is old, then close and remove from the list.
-                        if (timePassed.Milliseconds > HttpConstants.DefaultKeepAliveMilliseconds)
+                        count = m_ConnectedStreams.Count;
+                        for (int i = count - 1; i >= 0; i--)
                         {
-                            m_ConnectedStreams.RemoveAt(i);
-                            
-                            // Closes the socket to release resources.
-                            streamWrapper.Dispose();
+                            try
+                            {
+                                InputNetworkStreamWrapper streamWrapper = (InputNetworkStreamWrapper)m_ConnectedStreams[i];
+                                TimeSpan timePassed = curTime - streamWrapper.m_lastUsed;  // skigrinder - this is correct - original code was (m_lastUsed - curTime) - good evidence that persistent connections were never implemented here
+                                // If the socket is old, then close and remove from the list.
+                                if (timePassed.TotalMilliseconds > HttpConstants.DefaultKeepAliveMilliseconds)      // skigrinder - original code used timePassed.Milliseconds - need to use TotalMilliseconds  
```

Comment:
Thanks.  Are persistent connections a big priority?

Reply:
Don't recall these been asked before... so the answer is no, not a big priority. But if you can have those work, that would be a nice improvement.

Reply:
I'd eventually like to take a run at persistent connections.
This would probably improve http send/receive performance.
Can we commit/merge the existing fixes first?